tt_metal/api/command_queue.hpp

```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "types.hpp"
+
+#include "tt_metal/impl/buffers/buffer.hpp"
+
+//==================================================
+//                COMMAND QUEUE OPERATIONS
+//==================================================
+
+
+namespace tt::tt_metal{
+namespace v1 {
+
+/**
+ * @brief Reads a buffer from the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to read from.
+ * @param dst Pointer to the destination memory where data will be stored.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueReadBuffer(
+    CommandQueue cq,
+    Buffer buffer,
+    std::byte *dst,
+    bool blocking);
+
+/**
+ * @brief Writes data to a buffer on the device.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param buffer The device buffer to write to.
+ * @param src Source data vector to write to the device.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueWriteBuffer(
+    CommandQueue cq,
+    Buffer buffer,
+    const std::byte *src,
+    bool blocking);
+
+
+/**
+ * @brief Writes a program to the device and launches it.
+ *
+ * @param cq The command queue used to dispatch the command.
+ * @param program The program to execute on the device.
+ * @param blocking Indicates whether the operation is blocking.
+ */
+void EnqueueProgram(CommandQueue cq, Program program, bool blocking);
```

Comment:
What is the difference between this API and https://github.com/tenstorrent/tt-metal/pull/13759/files#diff-541cfc3156ac8f9d69e468ef5185b568636ad3a5e593ea4bda76055bab9d2c87R169

Reply:
Sounds like we do not want to officially support LaunchProgram but we do need to keep this for internal use.  I will move the declaration to another location.